Pages/Blog.php

```diff
@@ -84,6 +84,24 @@ public function get() {
             }
         }
 
+        foreach($blog->posts as $key => $post) {
+            //Header images
+            $header_image = '';
+
+            if(empty($post['header_image']) OR $post['header_image'] == '') {
+                preg_match_all('/<img\s+.*?src=[\"\']?([^\"\' >]*)[\"\']?[^>]*>/i',$post['body'],$matches,PREG_SET_ORDER);
+                if(!empty($matches[0][1])) {
+                    $header_image = (file_exists(HOME_PATH.$matches[0][1]))?$matches[0][1]:NULL;
+                } else {
+                    $header_image = NULL;
+                }
+            } else {
+                $header_image = (file_exists(HOME_PATH.'/img/blog/'.$post[header_image]))?'/img/blog/'.$post[header_image]:NULL;
+            }
+
+            $blog->posts[$key]['default_header_image'] = $header_image;
```

Comment:
this should just set the header_image - it's not default if it's getting set as the original custom image. also it should just set $post['header_image'] and your foreach should say foreach ($blog->posts as &$post)